### PR TITLE
Change node_id in runtime_context to hex string

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Instead of exposing the cython NodeID object, we want to standardize on the hex string for public node id.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
This breaks horovod which expects node_id to be NodeID instead of hex string.

**Review by ericl — APPROVED** [review]

File: python/ray/util/tracing/tracing_helper.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def _actor_hydrate_span_args(class_: _nameable, method: _nameable):
         "ray.function": f"{class_}.{method}",
         "ray.pid": str(os.getpid()),
         "ray.job_id": runtime_context["job_id"].hex(),
```
Comment (ericl) [inline python/ray/util/tracing/tracing_helper.py:246]: Actually, can we also do it for job_id (or any other Cython ids here)?

Comment (jjyao) [reply python/ray/util/tracing/tracing_helper.py:246]: Yes, I do plan to do the same thing for other ids but in follow-up PRs. Does that work for you?

Comment (ericl) [reply python/ray/util/tracing/tracing_helper.py:246]: Sure

**Review by pcmoritz — APPROVED** [review]
LGTM! Since there is no immediate urgency (and we have resolved the NodeAffinitySchedulingStrategy question), we can hold off merging this until after master is becoming the Ray 2.0 branch?

**Comment by jjyao** [comment]
Reopen it since we still want to do it once master becomes the 2.0 branch.

**Comment by ericl** [comment]
Sounds good. Rebase to pick up new changes in master? I'm not sure I trust test results this old to make a merge decision.

**Comment by jjyao** [comment]
> Sounds good. Rebase to pick up new changes in master? I'm not sure I trust test results this old to make a merge decision.

Yup, will do. We still cannot merge now since we haven't cut the 1.x branch yet.

**Comment by ericl** [comment]
Superceded by https://github.com/ray-project/ray/pull/26116

**Pull request closed without merging**